src/components/FeatureViews/Courses/CourseManagementContainer.js

```diff
@@ -1,463 +1,638 @@
-      (course) =>
-        checkFilter(course.academicLevel, gradeFilterValue) &&
-        checkFilter(course.courseCategory.id, subectFilterValue) &&
-        checkFilter(course.instructor.user.id, instructorsFilterValue)
-    )
-    .sort(
-      (firstEl, secondEl) =>
-        ({
-          start_date: sortDescOrder(firstEl.startDate, secondEl.startDate),
-          class_name: sortDescOrder(firstEl.title, secondEl.title),
-        }[sortByDate])
+    const {
+        data: studentData,
+        loading: studentLoading,
+        error: studentError,
+    } = useQuery(GET_STUDENTS, {
+        variables: { accountId },
+        skip: accountInfo.accountType !== 'PARENT',
+    });
+
+    if (courseLoading || studentLoading) return <Loading />;
+    if (courseError) return console.error(courseError.message);
+    if (studentError) return console.error(studentError.message);
+
+    const createFilteredListFromCourses = (filterCondition) =>
+        courseData.courses.reduce(
+            (accumulator, currentValue) =>
+                !accumulator.some((course) =>
+                    filterCondition(currentValue, course)
+                )
+                    ? [...accumulator, currentValue]
+                    : accumulator,
+            []
+        );
+    const subjectList = createFilteredListFromCourses(
+        (currentValue, course) =>
+            currentValue.courseCategory.id === course.courseCategory.id
+    );
+    const instructorsList = createFilteredListFromCourses(
+        (currentValue, course) =>
+            currentValue.instructor.user.id === course.instructor.user.id
     );
 
-  return (
-    <Grid item xs={12}>
-      <BackgroundPaper elevation={2}>
-        <Typography align="left" className="heading" variant="h3">
-          Course Management
-        </Typography>
-        <Paper elevation={4} className={classes.appBar}>
-          <Grid
-            container
-            alignItems="center"
-            className={classes.containerMargins}
-          >
-            <Grid item xs={3}>
-              <FormControl className={classes.margin}>
-                <Select
-                  labelId="course-management-sort-tab"
-                  id="course-management-sort-tab"
-                  displayEmpty
-                  value={sortByDate}
-                  onChange={handleChange}
-                  classes={{ select: classes.menuSelect }}
-                  input={<BootstrapInput />}
-                  MenuProps={{
-                    classes: { list: classes.dropdown },
-                    anchorOrigin: {
-                      vertical: "bottom",
-                      horizontal: "left",
-                    },
-                    transformOrigin: {
-                      vertical: "top",
-                      horizontal: "left",
-                    },
-                    getContentAnchorEl: null,
-                  }}
+
+    const checkFilter = (value, filter) => '' === filter || value === filter;
```

Comment:
what is this check? you're checking to see if filter is an empty string or equals to a given value? maybe a better name would be "isFilterValidValue" ?

Reply:
If it is an empty string that means there are no filters being set. So it will be true. (each state is set to an empty string) So after a user changes a filter, it will check for ```value.filter```.